todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/TasksTabletPresenter.java

```diff
@@ -0,0 +1,187 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.architecture.blueprints.todoapp;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.example.android.architecture.blueprints.todoapp.data.Task;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksDataSource;
+import com.example.android.architecture.blueprints.todoapp.data.source.TasksRepository;
+import com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailContract;
+import com.example.android.architecture.blueprints.todoapp.taskdetail.TaskDetailPresenter;
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksContract;
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksFilterType;
+import com.example.android.architecture.blueprints.todoapp.tasks.TasksPresenter;
+
+/**
+ * Presenter for the tablet screen that can act as a Tasks Presenter and a Task Detail Presenter.
+ */
+public class TasksTabletPresenter implements TasksContract.Presenter, TaskDetailContract.Presenter {
+
+    @NonNull
+    private final TasksRepository mTasksRepository;
+    @NonNull
+    private final TasksTabletNavigator mTasksNavigator;
+    @NonNull
+    private TasksPresenter mTasksPresenter;
+    @Nullable
+    private TaskDetailPresenter mTaskDetailPresenter;
+
+    public TasksTabletPresenter(@NonNull TasksRepository tasksRepository,
+            @NonNull TasksTabletNavigator tasksNavigator,
+            @NonNull TasksPresenter tasksPresenter) {
+        mTasksRepository = checkNotNull(tasksRepository);
+        mTasksNavigator = checkNotNull(tasksNavigator);
+        mTasksPresenter = checkNotNull(tasksPresenter);
+    }
+
+    @NonNull public TasksPresenter getTasksPresenter() {
+        return mTasksPresenter;
+    }
+
+    @Nullable public TaskDetailPresenter getTaskDetailPresenter() {
+        return mTaskDetailPresenter;
+    }
+
+    public void setTaskDetailPresenter(TaskDetailPresenter taskDetailPresenter) {
+        mTaskDetailPresenter = taskDetailPresenter;
+    }
+
+    /* TasksContract.Presenter methods can be called with or without a detail pane */
+
+    @Override
+    public void onTasksResult(int requestCode, int resultCode) {
+        mTasksPresenter.onTasksResult(requestCode, resultCode);
+    }
+
+    @Override
+    public void loadTasks(boolean forceUpdate) {
+        mTasksPresenter.loadTasks(forceUpdate);
+    }
+
+    @Override
+    public void addNewTask() {
+        mTasksPresenter.addNewTask();
+    }
+
+    @Override
+    public void openTaskDetails(@NonNull Task requestedTask) {
+        mTasksNavigator.startTaskDetailForTablet(requestedTask.getId());
```

Comment:
this is the only thing I'm not sure about, doesn't seem the `Navigator` responsibility to do this. Can we have a reference to the `Controller` instead?